src/shared/obj-bin-transform.js

```diff
@@ -0,0 +1,381 @@
+    systemFontInfoData.set(
+      new Uint8Array(this.#buffer, offset + 4, systemFontInfoLength)
+    );
+    return new FontInfo(systemFontInfoData.buffer);
+  }
+
+  static write(font) {
+    assert(
+      font.charProcOperatorList === undefined &&
+        font.defaultVMetrics === undefined &&
+        (font.fontExtraProperties === false ||
+          font.fontExtraProperties === undefined),
+      `FontInfo.write: Unsupported properties`
+    );
+    let offset = 0;
+    const buffer = new ArrayBuffer(10e4);
+    const data = new Uint8Array(buffer);
+    const view = new DataView(buffer);
+    const encoder = new TextEncoder();
+
+    const numBools = PROPS.bools.length;
+    let boolByte = 0,
+      boolBit = 0;
+    for (let i = 0; i < numBools; i++) {
+      const value = font[PROPS.bools[i]];
+      // eslint-disable-next-line no-nested-ternary
+      const bits = value === undefined ? 0x00 : value ? 0x02 : 0x01;
+      boolByte |= bits << boolBit;
+      boolBit += 2;
+      if (boolBit === 8 || i === numBools - 1) {
+        view.setUint8(offset++, boolByte);
+        boolByte = 0;
+        boolBit = 0;
+      }
+    }
+    assert(
+      offset === OFFSET_NUMBERS,
+      "FontInfo.write: Boolean properties offset mismatch"
+    );
+
+    for (const prop of PROPS.numbers) {
+      view.setFloat64(offset, font[prop]);
+      offset += 8;
+    }
+    assert(
+      offset === OFFSET_BBOX,
+      "FontInfo.write: Number properties offset mismatch"
+    );
+
+    if (font.bbox) {
+      view.setUint8(offset++, 4); // Number of coords
+      for (const coord of font.bbox) {
+        view.setInt16(offset, coord, true);
+        offset += 2;
+      }
+    } else {
+      view.setUint8(offset++, 0); // No BBox
+      offset += 2 * 4; // TODO: optimize this padding away
+    }
+
+    assert(
+      offset === OFFSET_FONT_MATRIX,
+      "FontInfo.write: BBox properties offset mismatch"
+    );
+
+    if (font.fontMatrix) {
+      view.setUint8(offset++, 6); // Number of points
+      for (const point of font.fontMatrix) {
+        view.setFloat32(offset, point, true);
+        offset += 4;
+      }
+    } else {
+      view.setUint8(offset++, 0); // No FontMatrix
+      offset += 4 * 6; // TODO: optimize this padding away
+    }
+
+    assert(
+      offset === OFFSET_STRINGS,
+      "FontInfo.write: FontMatrix properties offset mismatch"
+    );
+
+    view.setUint32(OFFSET_STRINGS, 0); // Placeholder for strings length
+    offset += 4;
+    for (const prop of PROPS.strings) {
+      // encodeInto doesn't work on SharedArrayBuffers or resizable ArrayBuffers
+      const encoded = encoder.encode(font[prop]);
+      view.setUint32(offset, encoded.length);
+      data.set(encoded, offset + 4);
+      offset += 4 + encoded.length;
+    }
+    view.setUint32(OFFSET_STRINGS, offset - OFFSET_STRINGS - 4);
+
+    if (font.systemFontInfo === null || font.systemFontInfo === undefined) {
```

Comment:
nit: here and for similar checks: `if (!font.systemInfo)...`

Reply:
Thanks